ursa.ino

```diff
@@ -107,14 +125,25 @@ void loop() {//on core 1. the balencing control loop will be here, with the goal
   if (tipped) {
     robotEnabled = false;//if the robot falls over, instead of running at full speed trying to right itself as it skids along the ground, it just disables
   }
+  if (recvdNewData) {
+    if (xSemaphoreTake(mutexRecv, 0) == pdTRUE) {
+      wifI = 0;
+      parseDataRecvd();
+      wifI = 0;

```

Comment:
Does `wifI` actually need to be a gobal variable? if it's going to be initialized every time this function is called I think you should just pass it to the parse functions

Reply:
The simplest way I could think of to get `wifI` into the parse and send functions and let the functions be able to increment `wifi` was to make a global variable. is really bad?

Reply:
First off, `wifI` isn't used in the loop function so it doesn't need to be used there. Instead, you could initialize a counter variable in the functions you call and then add a reference argument (`byte someFunction(byte data, byte& wifI` but with a better name for `wifI`)

Reply:
I was trying to make functions with pointers, but couldn't do it, so I switched to a global variable because I could do it simply. @circuitbird if you can make new functions, I'd be happy to change.

Reply:
fixed with 54d379a34b64520a469a093295422432225b8a77

Reply:
nice work! it does make github think you deleted then added back all the code and make merging mad